From b7fbcb6a5586422446cc357443b04216fae1bcda Mon Sep 17 00:00:00 2001
From: Iago Toral Quiroga <itoral@igalia.com>
Date: Fri, 22 Jul 2016 13:36:25 +0200
Subject: [PATCH 01/34] i965/disasm: also print nibctrl in IVB for execsize=8

4-wide DF operations where NibCtrl applies require and execsize of 8
in IvyBridge/BayTrail.

v2:
- Refactor NibCtrl printing (Matt)

Reviewed-by: Matt Turner <mattst88@gmail.com>
---
 src/mesa/drivers/dri/i965/brw_disasm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/mesa/drivers/dri/i965/brw_disasm.c b/src/mesa/drivers/dri/i965/brw_disasm.c
index 167067a4b7d..b04325564a6 100644
--- a/src/mesa/drivers/dri/i965/brw_disasm.c
+++ b/src/mesa/drivers/dri/i965/brw_disasm.c
@@ -1189,10 +1189,10 @@ qtr_ctrl(FILE *file, const struct gen_device_info *devinfo, brw_inst *inst)
 {
    int qtr_ctl = brw_inst_qtr_control(devinfo, inst);
    int exec_size = 1 << brw_inst_exec_size(devinfo, inst);
+   const unsigned nib_ctl = devinfo->gen < 7 ? 0 :
+                            brw_inst_nib_control(devinfo, inst);
 
-   if (exec_size < 8) {
-      const unsigned nib_ctl = devinfo->gen < 7 ? 0 :
-                               brw_inst_nib_control(devinfo, inst);
+   if (exec_size < 8 || nib_ctl) {
       format(file, " %dN", qtr_ctl * 2 + nib_ctl + 1);
    } else if (exec_size == 8) {
       switch (qtr_ctl) {
-- 
2.11.0

